src/test/unit/model/indexing/rvalue_cl_test.cpp

```diff
@@ -0,0 +1,247 @@
+#ifdef STAN_OPENCL
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+#include <stan/model/indexing/rvalue.hpp>
+#include <stan/model/indexing/rvalue_cl.hpp>
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <test/unit/model/indexing/util_cl.hpp>
+#include <tuple>
+
+using stan::model::rvalue;
+
+using stan::model::index_max;
+using stan::model::index_min;
+using stan::model::index_min_max;
+using stan::model::index_multi;
+using stan::model::index_omni;
+using stan::model::index_uni;
+
+template <typename T_eig, typename T_cl>
+void expect_eq(const T_eig& a, const T_cl& b) {
+  EXPECT_MATRIX_EQ(a, b);
+}
+void expect_eq(const double a, const double b) { EXPECT_EQ(a, b); }
+void expect_eq(const int a, const int b) { EXPECT_EQ(a, b); }
+
+TEST(ModelIndexing, rvalue_opencl_vector_1d) {
+  Eigen::VectorXd m(4);
+  m << 1, 2, 3, 4;
+  stan::math::matrix_cl<double> m_cl(m);
+  Eigen::VectorXi m_i(4);
+  m_i << 1, 2, 3, 4;
+  stan::math::matrix_cl<int> m_i_cl(m_i);
+  auto indices = std::make_tuple(
+      index_omni(), index_multi(std::vector<int>{1, 2, 1, 3, 1}), index_min(2),
+      index_max(3), index_min_max(2, 3), index_min_max(3, 1));
+  constexpr int N_ind = std::tuple_size<decltype(indices)>::value;
+  stan::math::index_apply<N_ind>([&indices, &m, &m_cl, &m_i,
+                                  &m_i_cl](auto... Is1) {
```

Comment:
Can you swap these for for_each() as well?

Reply:
done